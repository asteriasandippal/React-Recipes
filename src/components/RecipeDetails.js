import React from 'react';
import PropTypes from 'prop-types';

function RecipeDetails({ style, currentRecipe, className }) {
  if (!currentRecipe) {
    return (
      <div
        style={style}
        className={className}
      >
        <p className="h3 p2 bg-white italic center">
            Please select a recipw to see the details.
        </p>
      </div>
    );
  }
  return (
    <div style={style} className={className}>
      <div className="p2 bg-white">
        <h2 className="h2">{currentRecipe.name}</h2>
        <img
          alt=""
          className="fit"
          src={currentRecipe.image}
        />
        <div>
          <span>{currentRecipe.category}</span>
          <span>{currentRecipe.calories}</span>
        </div>
        <h3>RecipeIngredient</h3>
        {currentRecipe.ingredients &&
        <ul>
          {currentRecipe.ingredients.map(ingredient => (
            <li key={ingredient}>
              {ingredient}
            </li>
          ))}
        </ul>}

        <h3>Steps</h3>
        {currentRecipe.steps &&
          <ol>
            {currentRecipe.steps.map(step => (
              <li key={step}>
                {step}
              </li>
            ))}
          </ol>
        }
      </div>
    </div>
  );
}

RecipeDetails.defaultProps = {
  style: {},
  currentRecipe: '',
  className: '',
};

RecipeDetails.propTypes = {
  style: PropTypes.object,
  currentRecipe: PropTypes.object,
  className: PropTypes.string,
};

export default RecipeDetails;
