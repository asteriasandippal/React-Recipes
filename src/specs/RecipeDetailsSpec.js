import React from 'react';
import renderer from 'react-test-renderer';
// import BrowserRouter from 'react-router-dom';
import RecipeDetails from '../components/RecipeDetails';

describe('<RecipeDetails/>', () => {
  let testRecipe;
  beforeEach(() => {
    testRecipe = {
      id: 1,
      name: 'Test Recipe',
      category: 'Test Category',
      ingredients: ['Ing 1', 'Ing 2'],
      steps: ['Step 1', 'Step 2'],
    };
  });
  test('Should render zero state', () => {
    const component = renderer.create(<RecipeDetails currentRecipe={testRecipe} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should correctly asign a className', () => {
    const component = renderer.create(<RecipeDetails currentRecipe={testRecipe} className="classname-test" />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render recipe without ingredients', () => {
    delete testRecipe.ingredients;
    const component = renderer.create(<RecipeDetails currentRecipe={testRecipe} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render recipe without steps', () => {
    delete testRecipe.steps;
    const component = renderer.create(<RecipeDetails currentRecipe={testRecipe} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
