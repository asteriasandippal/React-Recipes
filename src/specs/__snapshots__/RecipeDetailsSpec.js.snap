// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetails/> Should correctly asign a className 1`] = `
<div
  className=""
  style={Object {}}
>
  <p
    className="h3 p2 bg-white italic center"
  >
    Please select a recipw to see the details.
  </p>
</div>
`;

exports[`<RecipeDetails/> Should render recipe without ingredients 1`] = `
<div
  className=""
  style={Object {}}
>
  <p
    className="h3 p2 bg-white italic center"
  >
    Please select a recipw to see the details.
  </p>
</div>
`;

exports[`<RecipeDetails/> Should render recipe without steps 1`] = `
<div
  className=""
  style={Object {}}
>
  <p
    className="h3 p2 bg-white italic center"
  >
    Please select a recipw to see the details.
  </p>
</div>
`;

exports[`<RecipeDetails/> Should render zero state 1`] = `
<div
  className=""
  style={Object {}}
>
  <p
    className="h3 p2 bg-white italic center"
  >
    Please select a recipw to see the details.
  </p>
</div>
`;

exports[`<RecipeDetails/> Should render zero state 2`] = `
<div
  className=""
  style={Object {}}
>
  <p
    className="h3 p2 bg-white italic center"
  >
    Please select a recipw to see the details.
  </p>
</div>
`;
